src/routes/LPOptimisers/EcosystemDeposit/DepositForm/MaturityDistribution/MaturityDistributionEntry/MaturityDistributionEntry.tsx

```diff
@@ -0,0 +1,90 @@
+import { DateTime } from 'luxon';
+import React, { useState } from 'react';
+
+import { IconLabel } from '../../../../../../components/composite/IconLabel/IconLabel';
+import { doNothing } from '../../../../../../utilities/doNothing';
+import {
+  DistributionInput,
+  DistributionInputWrapper,
+  EvenBox,
+  MaturityDistributionBox,
+  MaturityTypography,
+  PoolInputLabel,
+  PoolList,
+} from './MaturityDistributionEntry.styled';
+type MaturityDistributionEntryProps = {
+  distribution: number;
+  onChange: (value: number) => void;
+  maturityTimestamp: number;
+  pools: string[];
+  disabled: boolean;
+};
+export const MaturityDistributionEntry: React.FunctionComponent<MaturityDistributionEntryProps> = ({
+  distribution,
+  onChange = doNothing,
+  disabled,
+  maturityTimestamp,
+  pools,
+}) => {
+  const poolsCount = pools.length;
+  const [value, setValue] = useState(distribution.toString());
+  const handleOnBlur = (nextValue: number) => {
+    onChange(nextValue);
+    setValue(nextValue.toString());
+  };
+  return (
+    <MaturityDistributionBox>
+      <EvenBox>
+        <DistributionInputWrapper>
+          <DistributionInput
+            disabled={disabled}
+            max="100"
+            min="0"
+            type="number"
+            value={value}
```

Comment:
Just a suggestion which is out of scope for this version but maybe it would be good to write the logic here such that when a user changes the default value the other weights populate automatically. For example, in a 2 maturity card I would change default 50% to 70% and the other value should change to 30%. In a pool with 4 maturities if I change the first weight to 70% then the others change to 10%, 10%, 10%. If I then change the second weight to 20% the last two weights change to 5%, 5%. Wdyt?

Reply:
Hey @Sol3iman sounds as great suggestion for improvement. To me this seems as product decision and I would scope this idea in Notion and assign it to Renato for review. Then if agreed that it makes sense by him as well, we try to implement it 🙏

Reply:
Let's wait for Renato's opinion and we can incorporate this change separately. I'll mark it as resolved for now.